src/utransport.rs

```diff
@@ -27,37 +27,53 @@ use zenoh::{
 };
 
 #[inline]
-fn error_callback(
-    sender: &Sender<CallbackChannelMessage>,
-    listener: &Arc<dyn UListener>,
-    msg: &str,
-) {
-    log::error!("{msg}");
-    if sender
-        .send(CallbackChannelMessage {
-            listener: listener.clone(),
-            result: Err(UStatus::fail_with_code(UCode::INTERNAL, msg)),
-        })
-        .is_err()
-    {
-        log::error!("Unable to call the user callback");
+fn invoke_block_callback(listener: &Arc<dyn UListener>, resp_msg: Result<UMessage, &str>) {
+    match resp_msg {
+        Ok(umsg) => {
+            task::block_in_place(|| {
+                Handle::current().block_on(listener.on_receive(umsg));
```

Comment:
tokio is so much more explicit! Interesting.

So we need to grab a handle to the tokio Runtime running this function so that we can call block_on.

Reply:
Correct. In async_std, we don't need to specify runtime and can use block_on directly. However, in tokio, we can't call nested block_on. It forces us to use block_in_place and specify the handle explicitly.